lib/rules/attributes-order.js

```diff
@@ -0,0 +1,502 @@
+import { builders as b } from 'ember-template-recast';
+
+import createErrorMessage from '../helpers/create-error-message.js';
+import isValidConfigObjectFormat from '../helpers/is-valid-config-object.js';
+import replaceNode from '../helpers/replace-node.js';
+import Rule from './_base.js';
+
+const ERROR_MESSAGE = 'Your config does not match the allowed values';
+
+const AttributeType = {
+  ARGUMENTS: 'arguments',
+  ATTRIBUTES: 'attributes',
+  MODIFIERS: 'modifiers',
+  SPLATTRIBUTES: 'splattributes',
+};
+
+const DEFAULT_CONFIG = {
+  alphabetize: true,
+  attributeOrder: [AttributeType.ARGUMENTS, AttributeType.ATTRIBUTES, AttributeType.MODIFIERS],
+};
+
+export function createAttributesOrderErrorMessage(config) {
+  return createErrorMessage('attributes-order', ERROR_MESSAGE, config);
+}
+
+export default class AttributesOrder extends Rule {
+  appliedOrder = undefined;
+
+  parseConfig(config) {
+    const errorMessage = createAttributesOrderErrorMessage(config);
+    if (!isValidConfigObjectFormat(config, DEFAULT_CONFIG)) {
+      throw new Error(errorMessage);
+    }
+
+    if (config && typeof config === 'object') {
+      return { ...DEFAULT_CONFIG, ...config };
+    }
+
+    return DEFAULT_CONFIG;
```

Comment:
I think the case of `config` being `false` needs to be handled here (like [here](https://github.com/ember-template-lint/ember-template-lint/blob/master/lib/rules/attribute-indentation.js#L506)).

Also, the `config && typeof config === 'object'` could just be `typeof config === 'object')`:
```suggestion
    if (config === false) {
      return false;
    }

    if (typeof config === 'object') {
      return { ...DEFAULT_CONFIG, ...config };
    }

    return DEFAULT_CONFIG;
```